persist/src/main/java/com/codeforcommunity/database/tableImpl/DBImpl.java

```diff
@@ -0,0 +1,67 @@
+package com.codeforcommunity.database.tableImpl;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.time.format.DateTimeFormatter;
+import java.util.Properties;
+
+/**
+ * Our DBImpl class which {@link PostTableDBImpl} and {@link CommentTableDBImpl} will extend from.
+ * This is here to provide a few common functions that will be used. It has package-default
+ * visibility Since only the DBImpls will need to know about it.
+ */
+abstract class DBImpl {
+  private final String url;
+  private final String user;
+  private final String password;
+
+  /**
+   * The constructor which just calls the {@link DBImpl} super constructor.
+   *
+   * @param dbProperties A {@link Properties} we expect to contain the values for url, user, and
+   *     password for connecting to the database.
+   */
+  public DBImpl(Properties dbProperties) {
+    // Get the url, user, and password values from the provided properties object and make sure
+    // they're not null.
+    this.url = dbProperties.getProperty("database.url");
+    this.user = dbProperties.getProperty("database.username");
+    this.password = dbProperties.getProperty("database.password");
+
+    // Make sure nothing is null, otherwise this will cause issues when connecting.
+    if (this.url == null) {
+      throw new IllegalArgumentException("Database URL cannot be null.");
```

Comment:
I doubt this will matter but would an IllegalArgument be the appropriate exception to throw? Would an IllegalStateException be better?

Reply:
I'm going to leave it as illegal argument since it's checking arguments passed into a constructor